src/strongbox/github_api.clj

```diff
@@ -7,24 +7,188 @@
+  "returns the first non-nil value that can be used as a 'version' from a list of good candidates.
+  `asset` is a subset of `release` that has been filtered out from the other assets in the release.
+  ideally we want to use the name the author has specifically chosen for a release.
+  if that doesn't exist, we fallback to the git tag which is typically better than the asset's name."
+  [release map?, asset map?]
+  (let [;; most to least desirable
+        candidates [(:name release) (:tag_name release) (:name asset)]]
+    (->> candidates (map utils/nilable) (remove nil?) first)))
+
+(defn-spec parse-assets ::sp/list-of-maps
+  "filters and classifies a release's list of assets."
+  [release map?, known-game-tracks ::sp/game-track-list]
+  (let [supported-zip? #(-> % :content_type vector set (some supported-zip-mimes))
+        release-game-track (utils/guess-game-track (:name release))
+
+        release-n (:-i release)
+        latest-release? (= release-n 0)
+
+        release-json (->> release
+                          :assets
+                          (filter #(= "release.json" (:name %)))
+                          first)
+
+        published-before-classic? (some-> release
+                                          :published_at
+                                          (utils/dt-before? constants/release-of-wow-classic))
+
+        classify (fn [asset]
+                   (let [asset-game-track (utils/guess-game-track (:name asset))
+                         source-info {:game-track nil
+                                      :version (pick-version-name release asset)
+                                      :download-url (:browser_download_url asset)}
+
+                         game-track-list
+                         (cond
+                           ;; game track present in file name, prefer that over `:game-track-list` and any game-track in release name
+                           asset-game-track [asset-game-track]
+
+                           ;; I imagine there were classic addons published prior to it's release.
+                           ;; If we can use the asset name, brilliant, if not and it's before the cut off then it's retail.
+                           published-before-classic? [:retail]
+
+                           ;; game track present in release name, prefer that over `:game-track-list`
+                           release-game-track [release-game-track]
+
+                           ;; no game track present in asset name or release name and no `:game-track-list` BUT
+                           ;; a release.json file is present and this is still the first of N releases.
+                           ;; fetch release and match asset to it's contents.
+                           ;; if asset isn't found (!) then just use [nil]
+                           (and latest-release?
+                                release-json
+                                (empty? known-game-tracks))
```

Comment:
this is going to be almost all the time, effectively ignoring the release.json file. is this what we want?

Reply:
in fact - shift this check and using the known-game-tracks into the second pass.